File: app/src/main/java/com/example/task2/javafolder/viewviewmodel/fragments/BaseViewModel.java
package com.example.task2.javafolder.viewviewmodel.fragments;

import androidx.lifecycle.LiveData;
import androidx.lifecycle.MutableLiveData;
import androidx.lifecycle.ViewModel;

import com.example.task2.javafolder.model.ErrorName;
import com.example.task2.javafolder.repo.RepositoryImpl;

import javax.inject.Inject;

public abstract class BaseViewModel extends ViewModel {

    private final MutableLiveData<ErrorName> presenceOfErrorInUsersInput = new MutableLiveData<>();

    public final LiveData<ErrorName> presenceOfErrorInUsersInputLiveData = presenceOfErrorInUsersInput;

    @Inject
    public RepositoryImpl repository;

    public String getUsersInputForTextInput() {
        return String.valueOf(repository.getUsersInput());
    }

    protected void onEditTextChanged(String usersInputString) {
        presenceOfErrorInUsersInput.setValue(validateUsersInput(usersInputString));
    }

    protected ErrorName validateUsersInput(String input) {
        if (input.isEmpty()) {
            return ErrorName.EMPTY_INPUT;
        }
        try {
            int inputInteger = Integer.parseInt(input);
            if (inputInteger > 0) {
                repository.setUsersInput(inputInteger);
                return ErrorName.NO_ERROR;
            } else {
                return ErrorName.NEGATIVE_NUMBERS_INPUT;
            }
        } catch (NumberFormatException numberFormatException) {
            return ErrorName.INVALID_INPUT;
        }
    }

}

File: app/src/main/java/com/example/task2/kotlinfolder/viewviewmodel/fragments/BaseViewModel.kt
package com.example.task2.kotlinfolder.viewviewmodel.fragments

import androidx.lifecycle.LiveData
import androidx.lifecycle.MutableLiveData
import androidx.lifecycle.ViewModel
import com.example.task2.kotlinfolder.model.ErrorName
import com.example.task2.kotlinfolder.repo.Repository
import com.example.task2.kotlinfolder.repo.RepositoryImpl

abstract class BaseViewModel(protected val repository: Repository) : ViewModel() {
    private val presenceOfErrorInUsersInput: MutableLiveData<ErrorName> = MutableLiveData()

    internal val presenceOfErrorInUsersInputLiveData:
            LiveData<ErrorName> = presenceOfErrorInUsersInput

    private var isAlreadyInitialized = false

    fun attachView() {
        if (!isAlreadyInitialized) {
            isAlreadyInitialized = true
            onFirstAttach()
        }
    }

    protected open fun onFirstAttach() {}

    fun onEditTextChanged(usersInput: String) {
        presenceOfErrorInUsersInput.value = validateUsersInput(usersInput)
    }

    fun getUsersInputForTextInput(): String {
        return repository.usersInput.toString()
    }


    private fun validateUsersInput(input: String): ErrorName {
        if (input.isEmpty()) return ErrorName.EMPTY_INPUT
        return try {
            val inputInteger = input.toInt()
            if (inputInteger > 0) {
                repository.usersInput = inputInteger
                ErrorName.NO_ERROR
            } else {
                ErrorName.NEGATIVE_NUMBERS_INPUT
            }
        } catch (numberFormatException: NumberFormatException) {
            ErrorName.INVALID_INPUT
        }
    }
}